core/server/api_container/server/startosis_engine/plan_yaml/plan_yaml_generator.go

```diff
@@ -22,38 +22,56 @@ const (
 	outputFutureRefType    = "output"
 )
 
-// PlanYaml is a yaml representation of the effect of an Instructions Plan or sequence of instructions on the state of the Enclave.
-type PlanYaml struct {
-	privatePlanYaml *privatePlanYaml
+// PlanYamlGenerator is a yaml representation of the effect of an Instructions Plan or sequence of instructions on the state of the Enclave.
+type PlanYamlGenerator struct {
+	privatePlanYaml *PlanYaml
 
 	futureReferenceIndex map[string]string
 	filesArtifactIndex   map[string]*FilesArtifact
 	latestUuid           int
+	imageSet             map[string]bool
 }
 
-func CreateEmptyPlan(packageId string) *PlanYaml {
-	return &PlanYaml{
-		privatePlanYaml: &privatePlanYaml{
-			PackageId:      packageId,
-			Services:       []*Service{},
-			Tasks:          []*Task{},
-			FilesArtifacts: []*FilesArtifact{},
+func CreateEmptyPlan(packageId string) *PlanYamlGenerator {
+	return &PlanYamlGenerator{
+		privatePlanYaml: &PlanYaml{
+			PackageId:           packageId,
+			Services:            []*Service{},
+			Tasks:               []*Task{},
+			FilesArtifacts:      []*FilesArtifact{},
+			Images:              []string{},
+			PackageDependencies: []string{},
 		},
+		imageSet:             map[string]bool{},
 		futureReferenceIndex: map[string]string{},
 		filesArtifactIndex:   map[string]*FilesArtifact{},
 		latestUuid:           0,
 	}
 }
 
-func (planYaml *PlanYaml) GenerateYaml() (string, error) {
-	yamlBytes, err := yaml.Marshal(planYaml.privatePlanYaml)
+func (planYaml *PlanYamlGenerator) GenerateYaml(dependenciesOnly bool) (string, error) {
+	planYaml.AddImages()
```

Comment:
Puzzled by this.. are we adding empty images here? If so, why do we need to do this?